src/repository/postRepository.ts

```diff
@@ -4,17 +4,27 @@ import { db } from '../utils';
 
 class PostRepository {
 
+  // Selects the specifically wanted post by its id if found.
   async selectById(id: string): Promise<Post | undefined> {
     const query = 'select * from posts where id = ?';
     const [rows] = await db.pool.execute<RowDataPacket[]>(query, [id]);
-    if (rows[0]) return new Post(rows[0] as Post);
+    if (rows.length > 0) return new Post(rows[0] as Post);
   }
 
+  // Selects max 10 most recent posts. Where post's createdAt <= lastDate.
+  async selectRecent(lastDate: Date): Promise<Post[] | undefined> {
```

Comment:
It is enough to have `Promise<Post[]>` as the return type, since `undefined` is never actually returned (unlike when selecting only one post).